samples/Kconfig

```diff
@@ -16,15 +16,120 @@ config NCS_SAMPLES_DEFAULTS
+	bool
+	default y if BT_RPMSG_NRF53
+	imply NCS_ADD_CHILD_IMAGE_TO_DOMAIN
+
+config CHILD_IMAGE_802154_RPMSG
+	bool
+	default y if NRF_802154_SER_HOST
+	imply NCS_ADD_CHILD_IMAGE_TO_DOMAIN
+
+config CHILD_IMAGE_RPC_HOST
+	bool
+	default y if BT_RPC
+	imply NCS_ADD_CHILD_IMAGE_TO_DOMAIN
+
 config NCS_SAMPLE_EMPTY_APP_CORE_CHILD_IMAGE
 	bool "Enable empty_app_core as a child image"
 	default n
 	depends on SOC_NRF5340_CPUNET
-	select PARTITION_MANAGER_ENABLED
+	imply NCS_ADD_CHILD_IMAGE_TO_DOMAIN
 	help
 	  Add the empty_app_core as a child image for the given sample.
 	  Used by samples that only require SOC_NRF5340_CPUNET.
 
+config NCS_ADD_CHILD_IMAGE_TO_DOMAIN
+	bool
+	prompt "Add child image to the application/network core"
+	help
+	  Add child image to the other core. If the current sample is built
+	  for the APP core the child image will be build for the NET core and
+	  vice versa.
+
+if NCS_ADD_CHILD_IMAGE_TO_DOMAIN
+
+config NCS_DOMAIN_TO_ADD_CHILD_IMAGE_TO
+	string
+	default "CPUNET" if SOC_NRF5340_CPUAPP
+	default "CPUAPP" if SOC_NRF5340_CPUNET
+	help
+	  The name of the domain to which the child image will be added.
+
+config NCS_CHILD_IMAGE_NAME
+	# TODO should this be promptless when a sample is chosen automatically?
+	string
+	prompt  "Child image name"
+	default "empty_app_core" if CHILD_IMAGE_EMPTY_APP_CORE
+	default "multiprotocol_rpmsg" if CHILD_IMAGE_MULTIPROTOCOL_RPMSG
+	default "hci_rpmsg" if CHILD_IMAGE_HCI_RPMSG
+	default "802154_rpmsg" if CHILD_IMAGE_802154_RPMSG
+	default "rpc_host" if CHILD_IMAGE_RPC_HOST
+	help
+	  The name of the child image. This must match the name passed to the
+	  build strategy template.
+
+config NCS_CHILD_IMAGE_PATH
+	# TODO should this be promptless when a sample is chosen automatically?
+	string
+	prompt "Path to child image" 
+	default "${ZEPHYR_NRF_MODULE_DIR}/samples/nrf5340/empty_app_core" if CHILD_IMAGE_EMPTY_APP_CORE
+	default "${ZEPHYR_NRF_MODULE_DIR}/samples/nrf5340/multiprotocol_rpmsg" if CHILD_IMAGE_MULTIPROTOCOL_RPMSG
+	default "${ZEPHYR_BASE}/samples/bluetooth/hci_rpmsg" if CHILD_IMAGE_HCI_RPMSG
+	default "${ZEPHYR_BASE}/samples/boards/nrf/ieee802154/802154_rpmsg" if CHILD_IMAGE_802154_RPMSG
+	default "${ZEPHYR_NRF_MODULE_DIR}/samples/bluetooth/rpc_host" if CHILD_IMAGE_RPC_HOST
+	help
+	  Path to the child image to add to the other domain.
+
+config NCS_CHILD_IMAGE_OVERLAY
+	string "Child image overlay file"
+	default "${ZEPHYR_NRF_MODULE_DIR}/subsys/bluetooth/controller/bt_ll_softdevice_hci_rpmsg.conf" if BT_RPMSG_NRF53 && BT_LL_SOFTDEVICE_DEFAULT
+	help
+	  Add specified overlay file to the child image. Only one overlay file
+	  is supported. Compose a single overlay file wich is a combination
+	  of several files if the configuration of multiple overlay files is
+	  needed.
+
+# Add built strategies for all the automatic child images. Note that the module
+# name must match the upper case of the child image name.
+if CHILD_IMAGE_EMPTY_APP_CORE
+module=EMPTY_APP_CORE
```

Comment:
I want to avoid having to repeat the template inclusion, but I'm not able to find a way of having the value of `module` depend on what configuration is chosen. This becomes an issue when a customer points to their own sample as opposed to using one of the ones auto-included by us.

Reply:
resolved in the previous update